lib/cli-engine.js

```diff
@@ -300,18 +300,31 @@ function processFile(filename, configHelper, options) {
 
 /**
  * Returns result with warning by ignore settings
- * @param {string} filePath File path of checked code
- * @returns {Result} Result with single warning
+ * @param {string} filePath - File path of checked code
+ * @param {string} baseDir  - Absolute path of base directory
+ * @returns {Result}           Result with single warning
  * @private
  */
-function createIgnoreResult(filePath) {
+function createIgnoreResult(filePath, baseDir) {
+    var message;
+
+    if (/^\./.test(path.basename(filePath))) {
+        message = "Hidden file ignored by default.  Use '--ignore-pattern !.*' to override.";
```

Comment:
I was trying to keep this as simple as possible, knowing that the fancier I make it the more likely there will be bugs.  It's already pretty ugly, and hopefully can be cleaned up when we refactor to simplify default ignores.  For now, would you be opposed to a simpler message like:

```
File ignored by default.  Use a negated --ignore-pattern (like '!<filename>') to override."
```